libs/utils/android/workloads/camerapreview.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+
+from subprocess import Popen, PIPE
+from android import Screen, Workload
+from time import sleep
+from devlib.utils.android import adb_command
+
+import logging
+
+GET_FRAME_STATS_CMD = 'shell dumpsys gfxinfo {} > {}'
+
+class CameraPreview(Workload):
+    """
+    Android CameraPreview workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.GoogleCamera'
+    activity = 'com.android.camera.CameraActivity'
+
+    # Setup logger
+    logger = logging.getLogger('CameraPreview')
+    logger.setLevel(logging.INFO)
+
+
+    def __init__(self, test_env):
+        super(CameraPreview, self).__init__(test_env)
+        logging.debug('%14s - Workload created', 'CameraPreview')
+
+    def run(self, exp_dir, preview_duration_s, collect=''):
+
+        # Initialize energy meter results
+        nrg_data, nrg_file = None, None
+
+        # Unlock device screen (assume no password required)
+        self.target.execute('input keyevent 82')
+        # Press Back button to be sure we run the video from the start
+        self.target.execute('input keyevent KEYCODE_BACK')
+
+        # Force screen in LANDSCAPE mode
+        Screen.set_orientation(self.target, portrait=False)
+
+        # Start CameraPreview video on the target device
+        self.target.execute('am start -n {}/{}'.format(self.package,
+                                                       self.activity))
+        # Allow the activity to start
+        sleep(3)
+
+        # Reset framestats collection
+        self.target.execute('dumpsys gfxinfo {} reset'.format(self.package))
+
+        # Start energy collection
+        if 'energy' in collect and self.te.emeter:
+            self.te.emeter.reset()
+
+        logging.info("Preview camera for %d [s]", preview_duration_s)
+        sleep(preview_duration_s)
+
+        # Stop energy collection
+        if 'energy' in collect and self.te.emeter:
+            nrg_data, nrg_file = self.te.emeter.report(exp_dir)
+            logging.info("Estimated energy: %7.3f", float(nrg_data['BAT']))
+
+        # Get frame stats
+        db_file = os.path.join(exp_dir, "framestats.txt")
+        adb_command(self.target.adb_name,
+                    GET_FRAME_STATS_CMD.format(self.package, db_file))
```

Comment:
Why this adb_command call?
However... another candidate to pair with the previous System.collect_framestats(apk_name)...